### DISCUSSION TITLE
which router hook runs after dom updates?

### DISCUSSION THREAD

**Discussion by stefanprobst** [nuxt/nuxt#19301] [Questions]
i want to read `document.title` in a router callback (to send to analytics), but it seems that `router.afterEach` [runs before dom updates](https://router.vuejs.org/guide/advanced/navigation-guards.html#the-full-navigation-resolution-flow) - is there a way to safely read the updated `document.title` in a rotuer hook (other than wrapping in `setTimeout`)?

**Comment by Baroshem** [comment]
Hey,

What about `beforeRouteEnter ` hook with a check if an instance was already created (if it is an initial route change or either a true route change between the routes)?

**Reply by stefanprobst** [reply]
is `beforeRouteEnter` available in nuxt? i took [this](https://nuxt.com/docs/api/composables/use-router#navigation-guards) to mean that nuxt only provides `afterEach`, `beforeEach` and `beforeResolve`?

**Reply by AloisSeckar** [reply]
Since it is a Vue Router thing, it is surely awailable in Nuxt code too. But I know and tried only Options API syntax:

```
<script>
export default {
  beforeRouteEnter: function () {
    console.log('beforeRouteEnter');
  },
};
</script>
```

Demo here: https://stackblitz.com/edit/github-8cubd4
